Sources/NIO/SocketChannel.swift

```diff
@@ -331,6 +331,9 @@ final class DatagramChannel: BaseSocketChannel<Socket> {
 
     /// Support for vector reads, if enabled.
     private var vectorReadManager: Optional<DatagramVectorReadManager>
+    
+    /// The protocol family of this channel (if known)
+    private var protocolFamily: NIOBSDSocket.ProtocolFamily? = nil
```

Comment:
Is there any reason not to have this be computed from the `socket`?

Reply:
Now getting from socket.